pulp_rpm/app/tasks/publishing.py

```diff
@@ -479,25 +479,24 @@ def generate_repo_metadata(
     # work with select_related() and performs poorly with prefetch_related(). This is fine
     # because we know that Packages should only ever have one artifact per content.
     fields = [
-        "artifact__sha256",
-        "artifact__sha1",
-        "artifact__sha512",
-        "artifact__md5",
-        "artifact__sha224",
-        "artifact__sha384",
         "content_id",
         "content__rpm_package__checksum_type",
         "content__rpm_package__pkgId",
     ]
+    artifact_checksum = None
+    if package_checksum_type:
+        package_checksum_type = package_checksum_type.lower()
+        artifact_checksum = f"artifact__{package_checksum_type}"
```

Comment:
Maybe we should assert that the checksum type is 100% definitely one of the intended values, just to make absolutely sure that this can't be turned into SQL injection.

I'm pretty sure it can't already, but it wouldn't hurt.

Reply:
We have a check earlier https://github.com/pulp/pulp_rpm/blob/main/pulp_rpm/app/tasks/publishing.py#L445